syntax = "proto3";


service AsyncMarketServer{
  rpc SubQuoteStream(Symbols) returns (stream Quote);
  rpc SubAccount(AccountType) returns (FlagReply);
  rpc AddSubscribe(Symbols) returns (FlagReply);
  rpc StopEngine(FlagReply) returns (FlagReply);
}

message Symbols {
    repeated string symbols = 1;
}

message AccountType {
    string accounttype = 1;
}

message Quote{
  map<string, string> quote = 1;
}

message FlagReply {
    bool flag = 1;
}